lib/core/include/scipp/core/spatial_transforms.h

```diff
@@ -0,0 +1,194 @@
+// SPDX-License-Identifier: BSD-3-Clause
+// Copyright (c) 2021 Scipp contributors (https://github.com/scipp)
+/// @file
+/// @author Simon Heybrock
+#pragma once
+#include <Eigen/Core>
+#include <Eigen/Geometry>
+
+#include "scipp/core/dtype.h"
+
+namespace scipp::core {
+
+class RotationTransform {
+private:
+  // Store as quaterniond as this is more space efficient than storing as matrix
+  // (4 doubles for quat vs 9 doubles for 3x3 matrix).
+  Eigen::Quaterniond quat;
+
+public:
+  EIGEN_MAKE_ALIGNED_OPERATOR_NEW
+  RotationTransform() : quat(Eigen::Quaterniond::Identity()){};
+  explicit RotationTransform(const Eigen::Quaterniond &x) : quat(x){};
+
+  [[nodiscard]] Eigen::Matrix3d matrix() const {
+    return quat.toRotationMatrix();
+  }
+
+  inline bool operator==(const RotationTransform &other) const {
+    return this->matrix() == other.matrix();
+  }
```

Comment:
Is this implementation supposed to treat equivalent rotations as equal? I don't feel that that is a good default behavior for `operator==`? If we want to support checking for equivalence that should be done as a special explicit operation.

Here, I think we should just compare the coefficients.